### PR TITLE
jsdoc types processing added, refs #88.

### PR DESCRIPTION
**Pull request by Sozialarchiv**
With this pr a types file gets automatically generated during build.

### CODE REVIEW

File: package.json
Comment (neSpecc) [inline package.json:1]: It looks like the types aren't specified enough.
```
Could not find a declaration file for module '../editor.js/example/tools/list'. '../editor.js/example/tools/list/dist/list.umd.js' implicitly has an 'any' type.ts(7016)
```

Comment (neSpecc) [inline package.json:1]: also, please update a patch version

File: yarn.lock
Comment (neSpecc) [inline yarn.lock:1]: have you used newer version of yarn? Seems like there are too many changed

**Comment by Sozialarchiv** [comment]
If you think the types are not specific enough (It is admittedly more of a workaround) I see two alternatives:

1. Migrate the project to typescript. I would willing to create a pr.
2. Create a types file. Personally, I think the first part is technically cleaner (danger of divergence), but I also understand if the project should remain js.

**Comment by neSpecc** [comment]
> If you think the types are not specific enough (It is admittedly more of a workaround) I see two alternatives:
> 
> 1. Migrate the project to typescript. I would willing to create a pr.
> 2. Create a types file. Personally, I think the first part is technically cleaner (danger of divergence), but I also understand if the project should remain js.

Your workaround is fine. Just update package.json to specify types file. Check this example: https://github.com/editor-js/header/blob/master/package.json#L19-L25

File: yarn.lock
Comment (Sozialarchiv) [reply yarn.lock:1]: Which version should I use?

Comment (neSpecc) [reply yarn.lock:1]: 1.22

Comment (Sozialarchiv) [reply yarn.lock:1]: Now it should be done with yarn 1.22.22. But better check it again. Thanks for the quick feedback and the great editor.

**Comment by Sozialarchiv** [comment]
> Your workaround is fine. Just update package.json to specify types file. Check this example: https://github.com/editor-js/header/blob/master/package.json#L19-L25

A line with the types path should be  already included in package.json. Or am I misunderstanding something?
https://github.com/editor-js/list/blob/bfa3e388978fc9fa1cc887c83fdfe89741b34045/package.json#L22

**Comment by neSpecc** [comment]
> A line with the types path should be already included in package.json. Or am I misunderstanding something?
> 
> https://github.com/editor-js/list/blob/bfa3e388978fc9fa1cc887c83fdfe89741b34045/package.json#L22

You should also declare "types" property on the root level

**Comment by Sozialarchiv** [comment]
> You should also declare "types" property on the root level

Thanks a lot. I added it.

**Review by neSpecc — APPROVED** [review]
update a patch version, please

**Comment by Sozialarchiv** [comment]
> update a patch version, please

Done.

**Comment by Sozialarchiv** [comment]
Is there anything else to fix?

**Comment by neSpecc** [comment]
Thanks. Nice job. If you want to continue contributing, let me know

**Comment by Sozialarchiv** [comment]
@neSpecc 
> Thanks, @Sozialarchiv. Nice job. If you want to continue contributing, let me know

If you agree I would like to add this types-patch to the other core plugins without types. It is imho a pragmatic way to solve the missing types and it is wished by several people: https://github.com/codex-team/editor.js/discussions/2660 

In addition my plan is to solve some smaller issues which were complained about by our users. There are already pull requests for some of them. I will test and give feedback there.

**Comment by Sozialarchiv** [comment]
I have seen that some plugins are already written in typescript. Alternatively, I can also make typescript migrations if desired.

**Comment by neSpecc** [comment]
> @neSpecc
> 
> > Thanks, @Sozialarchiv. Nice job. If you want to continue contributing, let me know
> 
> If you agree I would like to add this types-patch to the other core plugins without types. It is imho a pragmatic way to solve the missing types and it is wished by several people: [codex-team/editor.js#2660](https://github.com/codex-team/editor.js/discussions/2660)
> 
> In addition my plan is to solve some smaller issues which were complained about by our users. There are already pull requests for some of them. I will test and give feedback there.

Let's add such patch to the [Table](https://github.com/editor-js/table) block.

**Comment by Sozialarchiv** [comment]
> Let's add such patch to the [Table](https://github.com/editor-js/table) block.

https://github.com/editor-js/table/pull/151

**Pull request merged**